### ISSUE TITLE
Add the ability to encode the mapping of the size of a symbol to MEI units

### ISSUE THREAD

**Issue by lpugin** [music-encoding/music-encoding#1061]
When using user defined symbols it is often necessary to give an indication how their size relates to the MEI unit. One current option is to give the dimension of the symbol in MEI units. For example:
```xml
<symbolDef width="2.4vu" height="3.4vu">
    <graphic mimetype="image/png" />
</symbolDef>
```
The disadvantage of this is that one need to calculate the dimensions. We could make this simpler by having a single attribute that provides the number of pixels per unit (e.g., `@ppu`). That way we do not need to have both the width and height values calculated
```xml
<symbolDef ppu="125">
    <graphic mimetype="image/png" />
</symbolDef>
```
Typically, for a glyph taken from a ~~SMuFL~~ [corr: Bravura] font, it would be `125` since the staff height is `1000`.

**Comment by ahankinson** [comment]
Trying to understand this:

From the [SMuFL documentation](https://w3c.github.io/smufl/latest/specification/scoring-metrics-glyph-registration.html)

"Dividing the em in four provides an analogue for a five-line staff: if a font uses 1000 upm (design units per em), as is conventional for a PostScript font, one staff space is equal to 250 design units; if a font uses 2048 upm, as is conventional for a TrueType font, one staff space is equal to 512 design units."

So it sounds like the `ppu` in your example comes from the PS font standard; if a font were a TrueType font, then `ppu` would be 256 to get the same appearance?

**Comment by lpugin** [comment]
You are right that SMuFL does not constraint the `upm` to be `1000`, so this is font dependant.

**Comment by lpugin** [comment]
Here is a concrete example where that would be useful. This is an external symbol from Bravura available in the SMuFL Browser (thank you @peterstadler for making this available):
https://smufl-browser.edirom.de/curlewSign.svg

One might want to use it and this is possible with a symbolDef. For example:
```xml
<symbolDef xml:id="curlew-sign">
   <graphic target="https://smufl-browser.edirom.de/curlewSign.svg" ulx="0" uly="0" />
</symbolDef>
```

However, we are missing some information about the size of the symbol in relation to other music elements. The size of the symbol in the SVG is based on the design unit in Bravura of 1000 per EM. We can calculate it 'by hand' in MEI units, (or in px if we have a determined number of pixels per MEI unit).

If we add these calculated values, and we can render it properly:
https://github.com/rism-digital/verovio.org/blob/1c5d3a0a295bf1b454882ebc85b01fee637de5a2/_tests/symboldef/symboldef-001.mei#L34-L36
![image](https://user-images.githubusercontent.com/689412/209134780-184331ba-1122-40f7-929c-3f02685c377f.png)

The proposal would be not to encoded the calculated values of `width` and `height`, but instead the mapping between the design units (here in Bravura) and the MEI units. 

Maybe `@ppu` is not ideal as a name? Other thoughts?

**Comment by ahankinson** [comment]
After some conversation with @lpugin: 

`@ppu` is a bit of a misnomer, since the intention of the attribute is to tell renderers how to translate some arbitrary measurement unit into an MEI unit. A generic way of thinking about it might be "external unit per MEI virtual unit." (For a quick refresher, an MEI "virtual unit" is a relative unit that describes half the distance between two staff lines; so a full five-line staff is 8vu.)

In the case of the examples provided above, the SVG was exported to a fixed size -- in this case, we have a size where 125 SMuFL "design units" are equal to 1 MEI vu. The mapping for the SMuFL export was 1 design unit = 1 pixel. 

However, there are use cases where we might want non-SMuFL symbols to be used. For example, an external PNG file that is 100px tall, and represents a note head. (Let's assume, for the sake of simplicity, that a note head occupies exactly two MEI virtual units and fits exactly between two staff lines.) In this case, the "external unit to MEI virtual unit" would be 50, since the external image is exactly two MEI virtual units tall. 

The other thing to consider is the definition of the [symbols themselves in MEI](https://music-encoding.org/guidelines/dev/content/shared.html#usersymbolsModuleUses). Although not particularly clear, it seems it is possible to define a symbol in MEI using MEI virtual units themselves, or simply some self-referential set of co-ordinates. This seems to point to the fact that, inside a `<symbolDef>` the co-ordinate space may be arbitrarily large (or small), but they will still need to correspond to a specific size when they are actually used. (For example, you may want two variants of the same symbol, one for "normal" and one for "cue" size). 

To illustrate, given this example from the section in the Guidelines: 

```xml
<symbolDef xml:id="userSymbols.triangleSymbol3">
    <line x="0" x2="2.55" y="0" y2="4.25"/>
    <line x="2.55" x2="5.1" y="4.25" y2="0"/>
    <line x="5.1" x2="0.85" y="0" y2="0"/>
</symbolDef>
```

We might assume an implicit "units per virtual unit" within this symbol of 1 -- thus all the measurements inside would be referring to virtual units on 1-to-1 basis. However, if we wanted to make a cue size out of this, we might do something like this:

```xml
<symbolDef xml:id="userSymbols.triangleSymbolCue" ppu="2">
    <symbol ref="#userSymbols.triangleSymbol3"/>
</symbolDef>
```

In which case we would reuse the existing definition, but simply draw it at half the size (2vu internal = 1vu external). 

This means that we would be able to map any arbitrary set of co-ordinates from things defined inside a `<symbolDef>` to the co-ordinate space in a given MEI file. So I think `@ppu` is a bit of a misdirection, since the external symbol may, in theory, be measured in non-pixel spaces (vu, em, pt, SMuFL design units, etc.). `@unitsPerVirtualUnit` is clear, but a bit of a mouthful.

How about `@vuMap`? `@vuMap="2"` would mean "map 2 (of some unit) per 1vu".

**Comment by lpugin** [comment]
I guess according to the attribute naming conventions in MEI it would be `@vumap`.
